cache_system/src/backend/policy/mod.rs

```diff
@@ -1065,6 +1379,29 @@ mod tests {
             // prevent SIGABRT due to double-panic
             if !std::thread::panicking() {
                 assert!(self.steps.is_empty(), "test steps left");
+                let mut tmp = TestSubscriberBackgroundTask::Invalid;
+                std::mem::swap(&mut tmp, &mut self.background_task);
+
+                match tmp {
+                    TestSubscriberBackgroundTask::NotStarted(_) => {
+                        // nothing to check
+                    }
+                    TestSubscriberBackgroundTask::Started(handle) => {
+                        // propagate panics
+                        if let Err(e) = handle.join() {
+                            if let Some(err) = e.downcast_ref::<&str>() {
+                                panic!("Error in background task: {err}")
+                            } else if let Some(err) = e.downcast_ref::<String>() {
+                                panic!("Error in background task: {err}")
+                            } else {
+                                panic!("Error in background task: <unknown>")
```

Comment:
I think `e` is display, so can't you do:

```suggestion
                                panic!("Error in background task: {}", e)
```

?

Reply:
This is not `std::error::Error`. It's actually `Box<dyn Any + Send>` which is NOT printable by default. Yeah, Rust allows you to panic with arbitrary types (although nobody does that we should always end up with `&str` or `String` (both can come out of `panic!(...)`).

Reply:
I vaguely remember having issues when someone panic'd with something that was Displayable but not `&str` or `&String` (`panic!()` with a format string actually called the panic hook with some crazy typed thing, but perhaps by the time the JoinHandle passes it back it will be a `&str` or `&String`)

More context

https://github.com/influxdata/influxdb_iox/pull/3130/files#r750597059

Reply:
This ain't a panic hook though. We don't have the actual backtrace / panic info at hand (that won't be `Send`). The join handle solely provides us with the value that was "raised" on the other thread.

In any case: this is solely for testing and shall provide a reasonable output. The more important part of this code snipped is to panic no matter what the panic payload on the other thread was so the test function panics and the test fails. This is tested by one of the "meta" tests.